pkg/crypto/wallet/wallet_test.go

```diff
@@ -0,0 +1,142 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package wallet
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/hyperledger/aries-framework-go/pkg/doc/did"
+	mockstorage "github.com/hyperledger/aries-framework-go/pkg/internal/mock/storage"
+	mockwallet "github.com/hyperledger/aries-framework-go/pkg/internal/mock/wallet"
+	"github.com/hyperledger/aries-framework-go/pkg/storage"
+)
+
+const (
+	serviceEndpoint = "sample-endpoint.com"
+)
+
+func TestBaseWallet_New(t *testing.T) {
+	t.Run("test error from OpenStore for keystore", func(t *testing.T) {
+		const errMsg = "error from OpenStore"
+		_, err := New(newMockWalletProvider(
+			&mockstorage.MockStoreProvider{ErrOpenStoreHandle: fmt.Errorf(errMsg)}))
+		require.Error(t, err)
+		require.Contains(t, err.Error(), errMsg)
+	})
+	t.Run("test error from OpenStore for did store", func(t *testing.T) {
+		_, err := New(newMockWalletProvider(
+			&mockstorage.MockStoreProvider{FailNameSpace: "didstore"}))
+		require.Error(t, err)
+		require.Contains(t, err.Error(), "failed to open store for name space")
+	})
+}
+
+func newMockWalletProvider(storagePvdr *mockstorage.MockStoreProvider) *mockProvider {
+	return &mockProvider{storagePvdr}
+}
+
+// mockProvider mocks provider for wallet
+type mockProvider struct {
+	storage *mockstorage.MockStoreProvider
+}
+
+func (m *mockProvider) StorageProvider() storage.Provider {
+	return m.storage
+}
+
+func (m *mockProvider) InboundTransportEndpoint() string {
+	return serviceEndpoint
+}
+
+func TestBaseWallet_Wrappers(t *testing.T) {
+	bw := BaseWallet{secretWallet: &mockwallet.CloseableWallet{
+		CreateEncryptionKeyValue: "encryption key",
+		CreateEncryptionKeyErr:   fmt.Errorf("err: CreateEncryptionKey"),
+		CreateSigningKeyValue:    "signing key",
+		CreateSigningKeyErr:      fmt.Errorf("err: CreateSigningKey"),
+		AttachCryptoOperatorErr:  fmt.Errorf("err: AttachCryptoOperator"),
+		ConvertKeyValue:          []byte("convert key"),
+		ConvertKeyErr:            fmt.Errorf("err: ConvertKey"),
+		FindVerKeyValue:          99,
+		FindVerKeyErr:            fmt.Errorf("err: FindVerKey"),
+		SignMessageValue:         []byte("signature"),
+		SignMessageErr:           fmt.Errorf("err: SignMessage"),
+		MockDID: &did.Doc{
+			ID: "mock doc id",
+		},
+	}}
+
+	t.Run("CreateEncryptionKey", func(t *testing.T) {
+		ret, err := bw.CreateEncryptionKey()
```

Comment:
It is not correct to to check return values from secret and wrapper wallet for equality. Create encryption key should always return different key.

Reply:
Same applies to tests bellow

Reply:
these tests use a mock wallet for `secretWallet`, so they can verify that wrapping methods are wrapping the correct methods of `secretWallet`. Functionality tests for the implementation are still present in `pkg/crypto/internal/wallet.SecretWallet`

Reply:
I've updated these tests to use the real SecretWallet instead of a mock secretWallet - these tests should confirm that the wrapper is correctly calling the wrapped functions.